R/apply_filters.R

```diff
@@ -87,3 +87,98 @@ apply_filters <- function(data,
+#'   set of PACTA data to be used in the analysis, based on the choice of an
+#'   allocation rule.
+#' @param start_analysis Numeric. A vector of length 1 indicating the start
+#'   year of the analysis.
+#' @param declining_technologies Character. A vector containing the technologies
+#'   defined as declining based on the scenario data.
+
+
+apply_filters <- function(data,
+                          investor = NULL,
+                          sectors = NULL,
+                          technologies = NULL,
+                          scenario_geography_filter = NULL,
+                          scenarios = NULL,
+                          allocation_method = NULL,
+                          start_analysis = NULL,
+                          declining_technologies) {
+  force(data)
+  investor %||% stop("Must provide input for 'investor'", call. = FALSE)
+  sectors %||% stop("Must provide input for 'sectors'", call. = FALSE)
+  technologies %||% stop("Must provide input for 'technologies'", call. = FALSE)
+  scenario_geography_filter %||% stop("Must provide input for 'scenario_geography_filter'", call. = FALSE)
+  scenarios %||% stop("Must provide input for 'scenarios'", call. = FALSE)
+  allocation_method %||% stop("Must provide input for 'allocation_method'", call. = FALSE)
+  start_analysis %||% stop("Must provide input for 'start_analysis'", call. = FALSE)
+
+  validate_data_has_expected_cols(
+    data = data,
+    expected_columns = c(
+      "year", "investor_name", "portfolio_name", "equity_market", "ald_sector",
+      "technology", "scenario", "allocation", "scenario_geography",
+      "plan_tech_prod", "plan_carsten", "scen_tech_prod", "plan_sec_prod",
+      "plan_sec_carsten"
+    )
+  )
+
+  if (!"id" %in% names(data)) {
+    data$id <- "PortfolioLevel"
+  }
+
+  if (!"company_name" %in% names(data)) {
+    data$company_name <- "PortfolioLevel"
+  }
+
+  data <- data %>%
+    dplyr::filter(
+      .data$ald_sector %in% .env$sectors,
+      .data$technology %in% .env$technologies,
+      .data$scenario_geography == .env$scenario_geography_filter,
+      .data$scenario %in% .env$scenarios,
+      .data$investor_name == .env$investor,
+      .data$allocation == .env$allocation_method,
+      !is.na(.data$plan_tech_prod)
+    )
+
+  remove <- data %>%
+    dplyr::filter(
+      .data$plan_tech_prod == 0,
+      .data$year == .env$start_analysis,
+      .data$technology %in% .env$declining_technologies
+    ) %>%
+    dplyr::mutate(
+      key = paste(.data$company_name, .data$id, .data$ald_sector, .data$technology, .data$scenario_geography)
+    ) %>%
+    # ADO 2393 - running distinct extracts the key as a variable with unique
+    # values to be used for removal of invalid entries below
+    dplyr::distinct(.data$key)
+
+  data <- data %>%
+    dplyr::mutate(
+      key = paste(.data$company_name, .data$id, .data$ald_sector, .data$technology, .data$scenario_geography)
+    ) %>%
```

Comment:
This overflows the 80 charactes (I think) and appears twice. Maybe write a small helper `paste_xyz_cols(data)` -- it  would remove the duplication and fit in a line.

Reply:
done